libraries/chain/eosio_contract.cpp

```diff
@@ -145,21 +146,44 @@ void apply_eosio_setcode(apply_context& context) {
    const auto& account = db.get<account_object,by_name>(act.account);
 
    int64_t code_size = (int64_t)act.code.size();
-   int64_t old_size  = (int64_t)account.code.size() * config::setcode_ram_bytes_multiplier;
+   int64_t old_size  = 0;
    int64_t new_size  = code_size * config::setcode_ram_bytes_multiplier;
 
    EOS_ASSERT( account.code_version != code_id, set_exact_code, "contract is already running this version of code" );
 
+   if(account.code_version != digest_type()) {
+      const code_object& old_code_entry = db.get<code_object, by_code_id>(account.code_version);
+      int64_t old_size  = (int64_t)old_code_entry.code.size();
```

Comment:
Should be `(int64_t)old_code_entry.code.size() * config::setcode_ram_bytes_multiplier`.